File: Leetcode/1726.cpp
class Solution {
public:
    int tupleSameProduct(vector<int>& nums) {
        int n = (int)nums.size();
        if (n < 4) return 0;
        unordered_map<int,int> unmap;
        for (int i = 0; i < n-1; ++i) {
            for (int j = i+1; j < n; ++j) {
                ++unmap[nums[i]*nums[j]];
            }
        }
        int ans = 0;
        for (const auto &[k,v] : unmap) {
            if (v > 1) ans += (v*(v-1))<<2;
        }
        return ans;
    }
};

File: Leetcode/1726.go
const MIN int = 4

func tupleSameProduct(nums []int) int {
	if len(nums) < MIN {
		return 0
	} 
	var m map[int]int
	m = make(map[int]int)
	for i := 0; i < len(nums)-1; i++ {
		for j := i+1; j < len(nums); j++ {
			m[nums[i]*nums[j]]++
		}
	}
	var ans = 0
	for _, v := range m {
		if v > 1 {
			ans += (v*(v-1)) << 2
		}
	}
	return ans
}